File: server/models/business.model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require("../config/db");

const Business = sequelize.define(
    "businesses",
    {
        id: {
            type: DataTypes.BIGINT,
            primaryKey: true,
            autoIncrement: true,
        },
        business_name: {
            type: DataTypes.STRING,
            allowNull: false
        },
        logo: {
            type: DataTypes.STRING,
            allowNull: true
        },
        user_id: {
            type: DataTypes.BIGINT,
            allowNull: false
        },
        business_type: {
            type: DataTypes.STRING,
            allowNull: true
        },
        email: {
            type: DataTypes.STRING,
            allowNull: true
        },
        phone: {
            type: DataTypes.STRING,
            allowNull: true
        },
        upi_id: {
            type: DataTypes.STRING,
            allowNull: true
        },
        account_holder_name: {
            type: DataTypes.STRING,
            allowNull: true
        },
        account_number: {
            type: DataTypes.STRING,
            allowNull: true
        },
        ifsc_code: {
            type: DataTypes.STRING,
            allowNull: true
        },
        bank_name: {
            type: DataTypes.STRING,
            allowNull: true
        },
        sync_email: {
            type: DataTypes.STRING,
            allowNull: true
        },
        google_refresh_token: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        google_sync_enabled: {
            type: DataTypes.BOOLEAN,
            defaultValue: true
        },
        slug: {
            type: DataTypes.STRING,
            allowNull: true,
            unique: true
        },
        selected_template: {
            type: DataTypes.STRING,
            defaultValue: 'template1'
        },
        website_enabled: {
            type: DataTypes.BOOLEAN,
            defaultValue: false
        },
        has_multiple_locations: {
            type: DataTypes.BOOLEAN,
            defaultValue: false
        },
        address: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        city: {
            type: DataTypes.STRING,
            allowNull: true
        },
        state: {
            type: DataTypes.STRING,
            allowNull: true
        },
        location_type: {
            type: DataTypes.ENUM('Physical', 'Online'),
            defaultValue: 'Physical'
        },
        meeting_link: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        website_type: {
            type: DataTypes.ENUM('website', 'portfolio'),
            defaultValue: 'website'
        },
        description: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        status: {
            type: DataTypes.BOOLEAN,
            defaultValue: true
        },
        suspended_reason: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        created_at: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW
        },
        updated_at: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW
        }
    },
    {
        timestamps: false,
        tableName: "businesses",
        indexes: [
            { fields: ["user_id"] },
            { fields: ["slug"], unique: true },
            { fields: ["status"] }
        ]
    }
);

module.exports = Business;

File: my_booking_application/lib/app/data/models/business_model.dart
class BusinessModel {
  int? id;
  String? businessName;
  String? businessType;
  String? slug;
  String? address;
  String? city;
  String? state;
  bool? status;
  int? userId;

  BusinessModel({
    this.id,
    this.businessName,
    this.businessType,
    this.slug,
    this.address,
    this.city,
    this.state,
    this.status,
    this.userId,
  });

  BusinessModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    businessName = json['business_name'];
    businessType = json['business_type'];
    slug = json['slug'];
    address = json['address'];
    city = json['city'];
    state = json['state'];
    status = json['status'];
    userId = json['user_id'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['business_name'] = businessName;
    data['business_type'] = businessType;
    data['slug'] = slug;
    data['address'] = address;
    data['city'] = city;
    data['state'] = state;
    data['status'] = status;
    data['user_id'] = userId;
    return data;
  }
}
